### PR TITLE
Fuse Adam And SGD ops

### PR DESCRIPTION
**Pull request by chengduoZH**
Code separated from #15497

Transformer:

执行引擎 | GPU卡 | FuseAdam | 速度 | 加速比
-- | -- | -- | -- | --
threaded_executor | 1 | 否 | 4.162 | 1
threaded_executor | 1 | 是 | 4.27 | 0.026

### CODE REVIEW

File: paddle/fluid/framework/details/build_strategy.h [hunk 1]
```diff
@@ -74,8 +74,12 @@ struct BuildStrategy {
 
   bool fuse_elewise_add_act_ops_{false};
 
+  bool fuse_all_optimizer_ops_{false};
```
Comment (panyx0718) [inline paddle/fluid/framework/details/build_strategy.h:78]: enable by default?

Comment (panyx0718) [reply paddle/fluid/framework/details/build_strategy.h:78]: ?

Comment (chengduoZH) [reply paddle/fluid/framework/details/build_strategy.h:78]: I'm not sure if this pass can improve performance on all models, today I'll turn it on and run on CE.

Comment (panyx0718) [reply paddle/fluid/framework/details/build_strategy.h:78]: what's the result?

Comment (chengduoZH) [reply paddle/fluid/framework/details/build_strategy.h:78]: For some models, the speed can be increased by 19%, for example, the model_object_detection model (in the case of 4 cards), in some models, the speed decreases slightly by 1%.

**Review by panyx0718 — DISMISSED** [review]

**Review by panyx0718 — DISMISSED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**